docs/source/metrics.rst

```diff
@@ -196,12 +196,71 @@ Metric API
 .. autoclass:: pytorch_lightning.metrics.Metric
     :noindex:
 
-*************
-Class metrics
-*************
+***************************
+Class vs Functional Metrics
+***************************
 
+The functional metrics follow the simple paradigm input in, output out. This means, they don't provide any advanced mechanisms for syncing across DDP nodes or aggregation over batches. They simply compute the metric value based on the given inputs.
+
+Also the integration within other parts of PyTorch Lightning will never be as tight as with the class-based interface.
+If you look for just computing the values, the functional metrics are the way to go. However, if you are looking for the best integration and user experience, please consider also to use the class interface.
+
+**********************
 Classification Metrics
-----------------------
+**********************
+
+Input types
+-----------
+
+For the purposes of classification metrics, inputs (predictions and targets) are split 
+into these categories (``N`` stands for the batch size and ``C`` for number of classes):
+
+.. csv-table:: \*dtype ``binary`` means integers that are either 0 or 1
+    :header: "Type", "preds shape", "preds dtype", "target shape", "target dtype"
+    :widths: 20, 10, 10, 10, 10
+
+    "Binary", "(N,)", "``float``", "(N,)", "``binary``\*"
+    "Multi-class", "(N,)", "``int``", "(N,)", "``int``"
+    "Multi-class with probabilities", "(N, C)", "``float``", "(N,)", "``int``"
+    "Multi-label", "(N, ...)", "``float``", "(N, ...)", "``binary``\*"
+    "Multi-dimensional multi-class", "(N, ...)", "``int``", "(N, ...)", "``int``"
+    "Multi-dimensional multi-class with probabilities", "(N, C, ...) or (N, ..., C)", "``float``", "(N, ...)", "``int``"
```

Comment:
IMO we should only allow the first option: `(N, C, ...)`.
If this is specified in docs (like here, really like the table :] ) then it will up to the user to reshape

Reply:
I may agree here that all options would be quite confusing when code is going to be shared among users...

Reply:
Alright, if there seems to be a consensus on this I'll change it

Reply:
Let's ping also @justusschock and @teddykoker and @ananyahjha93

Reply:
yes, just (N,C,...) is what I'd expect as well, since this is the standard tensor format in pytorch here. If user's are using anything else on purpose they should take care of proper formatting by themselves.

Reply:
Cool, whill change then